src/custom-element.js

```diff
@@ -381,6 +381,9 @@ export function createAmpElementProto(win, name, opt_implementationClass) {
      * @private {boolean|undefined}
      */
     this.isInTemplate_ = undefined;
+
+    /** @private {boolean} */
+    this.hadLoadingError_ = false;
```

Comment:
I think this is definitely the right way to solve it, but we can make it a little smarter by using a counter (like `#layoutCount`) instead of a boolean. We can then check if `#layoutAttempts` (naming?) is `> 0` and refuse to show loading.